### PR TITLE
feat(fixit): Adding privateca/ca_pool samples

### PR DESCRIPTION
**Pull request by rsamborski**
## Description

Fixes b/281691685, b/281692014

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [X] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [X] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: privateca/ca_pool_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+func TestCaPoolTests(t *testing.T) {
+	var r *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	projectId := tc.ProjectID
+	location := "us-central1"
+	caPoolId := fmt.Sprintf("test-ca-pool-%v-%v", time.Now().Format("2006-01-02"), r.Int())
+
+	buf := &bytes.Buffer{}
```
Comment (telpirion) [inline privateca/ca_pool_test.go:35]: style: we typically just declare an empty buffer like so:

```
var buf *bytes.Buffer
```

File: privateca/ca_pool_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	if err := createCaPool(buf, projectId, location, caPoolId); err != nil {
+		t.Errorf("createCaPool got err: %v", err)
```
Comment (telpirion) [inline privateca/ca_pool_test.go:38]: issue:  replace `t.Errorf` with `t.Fatal`. If this test has failed, the rest of them will too. Also just print the error--there's no need to format it.

File: privateca/ca_pool_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	expectedResult := "CA Pool created"
+	if got := buf.String(); !strings.Contains(got, expectedResult) {
+		t.Errorf("createCaPool got %q, want %q", got, expectedResult)
+	}
+
+	buf.Reset()
+
+	if err := deleteCaPool(buf, projectId, location, caPoolId); err != nil {
```
Comment (telpirion) [inline privateca/ca_pool_test.go:48]: issue: create a separate test case for `deleteCaPool()`. We should have a 1-to-1 mapping of test cases to sample methods.

File: privateca/ca_pool_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		t.Errorf("deleteCaPool got err: %v", err)
```
Comment (telpirion) [inline privateca/ca_pool_test.go:49]: Same as above--replace `t.Errorf()` with `t.Fatal()`. If there is an err when the sample is called, the rest of sample will fail too.

File: privateca/ca_pool_test.go (hunk 3 above)
Comment (telpirion) [reply privateca/ca_pool_test.go:48]: Also, using samples to teardown other samples often backfires. I would recommend using the API directly in a dedicated teardown function.

File: privateca/create_ca_pool.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		return fmt.Errorf("NewCertificateAuthorityClient creation failed: %w", err)
+	}
+	defer caClient.Close()
+
+	caPool := &privatecapb.CaPool{
+		// Set the tier (see: https://cloud.google.com/certificate-authority-service/docs/tiers).
+		Tier: privatecapb.CaPool_ENTERPRISE,
+	}
+
+	locationPath := "projects/" + projectID + "/locations/" + location
```
Comment (telpirion) [inline privateca/create_ca_pool.go:46]: nit: use `fmt.Sprintf()` instead here.

```
locationPath := fmt.Sprintf("projects/%s/locations/%s", projectID, location)
```

File: privateca/create_ca_pool.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+	req := &privatecapb.CreateCaPoolRequest{
+		// See https://pkg.go.dev/cloud.google.com/go/security/privateca/apiv1/privatecapb#CreateCaPoolRequest.
```
Comment (telpirion) [inline privateca/create_ca_pool.go:49]: Move comment above this line.

File: privateca/delete_ca_pool.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	// caPoolId := "ca-pool-id"		// A unique id/name for the ca pool.
+
+	ctx := context.Background()
+	caClient, err := privateca.NewCertificateAuthorityClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewCertificateAuthorityClient creation failed: %w", err)
+	}
+	defer caClient.Close()
+
+	fullCaPoolName := "projects/" + projectID + "/locations/" + location + "/caPools/" + caPoolId
```
Comment (telpirion) [inline privateca/delete_ca_pool.go:41]: See previous--use `fmt.Sprintf()` here instead.

File: privateca/delete_ca_pool.go [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+	req := &privatecapb.DeleteCaPoolRequest{
+		// See https://pkg.go.dev/cloud.google.com/go/security/privateca/apiv1/privatecapb#CreateCaPoolRequest.
```
Comment (telpirion) [inline privateca/delete_ca_pool.go:44]: Move comment above line of code. Is this supposed to link to the `CreateCaPoolRequest` message?

Comment (rsamborski) [reply privateca/delete_ca_pool.go]: Good catch!

**Pull request merged**